stackstate_checks_base/stackstate_checks/base/utils/http_helper.py

```diff
@@ -0,0 +1,1447 @@
+        elif response.status_code >= 400:
+            errors.append("""Server responded with a {0}""".format(str(response.status_code)))
+
+        # If the HTTP Response Type has been set
+        # Test if the HTTP Response Type is a enum of type HTTPResponseType
+        if self._validate_type is not None and \
+                type(self._validate_type) is HTTPResponseType and \
+                isinstance(self._validate_type, Enum) is True:
+
+            # If the type set is JSON. We then attempt to parse it and see if it is valid
+            if self._validate_type is HTTPResponseType.JSON:
+                try:
+                    data = json.loads(response.content.decode(response.encoding))
+                    if isinstance(data, dict) is False:
+                        raise ValueError()
+
+                except ValueError as e:
+                    errors.append("""Unable to parse the response as JSON.
+                                     The response received was {0}.
+                                     Full error from the JSON parse attempt {1}
+                                     """.format(str(response.content), e))
+
+            elif isinstance(self._validate_type.value, type(response.content)) is False:
+                errors.append("""The response content type does not conform to the validation type. The response
+                                 type is {0}.
+                                 The expected type is {1}
+                                 To fix this you can either modify the type validation or remove it entirely
+                                 """.format(str(type(response.content)), str(self._validate_type.value)))
+
+        # Return all the errors that occurred
+        return errors
+
+
+"""
+    The HTTP Helper Handler is used to create compact methods using most of the function defined in the
+    Connection, Request, Session and Response Helpers
+    Functionality:
+        - Overwrite Connection Helper
+        - Overwrite Request Helper
+        - Overwrite Session Helper
+        - Overwrite Response Helper
+        - GET Request
+        - POST Request
+        - DELETE Request
+        - PATCH Request
+        - PUT Request
+
+    There is 3 ways to approach creating a HTTP Helper
+    - The developer can create the HTTPHelperRequestHandler, HTTPHelperSessionHandler, HTTPHelperResponseHandler and
+      HTTPHelperConnectionHandler.
+      These can then manually be passed down as kwargs to the send function
+    - The developer can create the HTTPHelperRequestHandler, HTTPHelperSessionHandler, HTTPHelperResponseHandler and
+      HTTPHelperConnectionHandler.
+      These can then manually be set within the HTTPHelper class with the overwrite functions
+    - Or it can be left to the internal state where the Helper control and build the state.
+      Thus if the developer use the setters and getter those internal state will be manipulated
+"""
+
+
+class HTTPHelper:
```

Comment:
I would place this at the top of the file. This is what you want people to use and read about.

Reply:
Sure will do